### ISSUE TITLE
HiveError: This should not happen. (during Hive.openBox)

### ISSUE THREAD

**Issue by rubenvereecken** [hivedb/hive#639]
**Steps to Reproduce**
User started the app. Then something bad happened. All we have is an automated bug report with stack trace. Our basic Hive initialization seemed to have run just fine. But right after that, it looks like while a box was being opened, this error occurred.

**Code sample**
```
> HiveError: This should not happen. Please open an issue on GitHub.
> #0   BinaryReaderImpl.readFrame (package:hive/src/binary/binary_reader_impl.dart:249)
> #1   FrameHelper.framesFromBytes (package:hive/src/binary/frame_helper.dart:17)
> #2   FrameIoHelper.framesFromFile (package:hive/src/io/frame_io_helper.dart:41)
> #3   <asynchronous suspension>
> #4   StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:82)
> #5   <asynchronous suspension>
> #6   HiveImpl._openBox (package:hive/src/hive_impl.dart:106)
> #7   <asynchronous suspension>
> #8   HiveImpl.openBox (package:hive/src/hive_impl.dart:135)
> #9   <asynchronous suspension>
> #10   Collection.initialize (package:woushe_refactored/src/models/data_models/collection.dart:79)
> #11   <asynchronous suspension>
> #12   Future.wait.<anonymous closure> (dart:async/future.dart)
> #13   <asynchronous suspension>
> #14   MeetupsRepository.initialize (package:woushe_refactored/src/repositories/meetups_repository.dart:55)
> #15   <asynchronous suspension>
> #16   Future.wait.<anonymous closure> (dart:async/future.dart)
> #17   <asynchronous suspension>
> #18   RepositoryBarrel.initializeAll (package:woushe_refactored/src/repositories/repository_barrel.dart:70)
> #19   <asynchronous suspension>
> #20   StartupScreenBloc.startup (package:woushe_refactored/src/screens/startup_screen_bloc.dart:84)
> #21   <asynchronous suspension>
> #22   
> 18:03:53.341 (+0:00:02.160626)
> On startup error
```

Here's the relevant part where the error seemed to have occurred:
`data_models/collection.dart:79`
```
  Future<void> initialize() async {
    box = await Hive.openBox<T>(name);
    refs = await Hive.openBox<List>('$name:refs');
  }
```

And here's our start-up sequence up until that point:
```
    await Hive.initFlutter(await LessgoPath.getDatabaseRelativePath());

    Hive.registerAdapter(UserAdapter());
    Hive.registerAdapter(UserSettingAdapter());
  // etc etc
```

**Version**
 - Platform: Android
 - Flutter version: 1.22.6
 - Hive version: 1.4.4+1

**Comment by anisalibegic** [comment]
On device SM-G991B and Android 11.

**Comment by codetinker** [comment]
Same here. Happen only on android 11
Other android versions tested and working no problem.

        [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: HiveError: This should not happen. Please open an issue on GitHub.
        E/flutter ( 5139): #0      BinaryReaderImpl.readFrame (package:hive/src/binary/binary_reader_impl.dart:250:7)
        E/flutter ( 5139): #1      FrameHelper.framesFromBytes (package:hive/src/binary/frame_helper.dart:21:26)
        E/flutter ( 5139): #2      FrameIoHelper.framesFromFile (package:hive/src/io/frame_io_helper.dart:42:12)
        E/flutter ( 5139): <asynchronous suspension>
        E/flutter ( 5139): #3      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:86:11)
        E/flutter ( 5139): <asynchronous suspension>
        E/flutter ( 5139): #4      HiveImpl._openBox (package:hive/src/hive_impl.dart:116:9)
        E/flutter ( 5139): <asynchronous suspension>
        E/flutter ( 5139): #5      HiveImpl.openBox (package:hive/src/hive_impl.dart:145:12)
        E/flutter ( 5139): <asynchronous suspension>
        E/flutter ( 5139): #6      GlobalBloc.gSave (package:app/provider.dart:639:17)
        E/flutter ( 5139): <asynchronous suspension>
        E/flutter ( 5139):

**Comment by rubenvereecken** [comment]
Just had it happen on Android again.

**Comment by alexshikov** [comment]
We get the same exception from time to time. I would appreciate any ideas on why this could happen and what workarounds we could try.

**Comment by rubenvereecken** [comment]
@alexshikov it only happens rarely so it doesn't seem to be a data migration issue or we would have it happen with more clients. The only thing that's worked for us is put a `catch` block in that deletes the whole Hive data folder and asks the user to restart the app.

**Comment by n13** [comment]
This sucks, I've also had this happen, and ended up writing a wrapper class called SafeHive, which catches the error on open, deletes the file, and creates a new file. 

In our case this is OK since it was just used for caching, so the cache has to be refilled. 

But this sort of instability has me considering alternatives to Hive. It seems it's a rare hard to trace bug, which is even worse (since it's hard / impossible to reproduce on purpose).

**Comment by vixez** [comment]
Having this issue as well, occurs randomly

**Comment by cspensky** [comment]
Also seeing this.. Struggling to track it down.

**Comment by themisir** [comment]
Closed with #914.

Please upgrade to hive [v2.0.6](https://pub.dev/packages/hive/versions/2.0.6).

**Comment by n13** [comment]
> Closed with #914.
> 
> Please upgrade to hive [v2.0.6](https://pub.dev/packages/hive/versions/2.0.6).

Wonderful, thank you! I read the PR that sounds like a pretty horrible bug to debug / find / reproduce. Well done and many thanks.
